packages/ckeditor5-ui/src/dialog/dialog.ts

```diff
@@ -88,7 +88,9 @@ export default class Dialog extends Plugin {
 		}, { priority: 'low' } );
 
 		this.on<DialogHideEvent>( 'hide', () => {
-			this._hide();
+			if ( Dialog._visibleDialogPlugin ) {
+				Dialog._visibleDialogPlugin._hide();
```

Comment:
AFAIU, the way currently implemented it leads to `hide` event being fired for _this_ `Dialog` instance, while _other_ `Dialog` instance is actually hidden. This may lead to errors if someone does something on hide.

Instead, `Dialog#hide()` should cause `hide` event be fired for `_visibleDialogPlugin` instance (same for `Dialog#show()` of course -- and right now this is not consistent, AFAICS).